dwave/cloud/config.py

```diff
@@ -48,171 +48,141 @@
 
 Environment variables:
 
-    ``DWAVE_CONFIG_FILE``: Configuration file path.
+*   ``DWAVE_API_CLIENT``: API client class. Supported values are ``qpu``, ``sw`` and ``hybrid``.
+*   ``DWAVE_API_ENDPOINT``: API endpoint URL.
+*   ``DWAVE_API_HEADERS``: Optional additional HTTP headers.
+*   ``DWAVE_API_PROXY``: URL for proxy connections to D-Wave API.
+*   ``DWAVE_API_REGION``: API region code.
+*   ``DWAVE_API_SOLVER``: Default solver.
+*   ``DWAVE_API_TOKEN``: API authorization token.
+*   ``DWAVE_CONFIG_FILE``: Configuration file path. 
+*   ``DWAVE_METADATA_API_ENDPOINT``: Metadata API endpoint URL.
+*   ``DWAVE_PROFILE``: Name of profile (section).
 
-    ``DWAVE_PROFILE``: Name of profile (section).
-
-    ``DWAVE_API_CLIENT``: API client class. Supported values are ``qpu``, ``sw`` and ``hybrid``.
+Examples:
+    The following are typical and advanced examples of using 
+    :func:`~dwave.cloud.client.Client.from_config` to create a configured client.
 
-    ``DWAVE_API_REGION``: API region code.
+    *   **Example** Typical use for QPU and hybrid solvers
 
-    ``DWAVE_API_ENDPOINT``: API endpoint URL.
+    This example uses a configuration file in one of the standard paths, 
+    :code:`~/.config/dwave/dwave.conf`, selected through auto-detection
+    on the local system following the user/system configuration paths of 
+    :func:`get_configfile_paths`, to provide the following default 
+    configuration::
 
-    ``DWAVE_METADATA_API_ENDPOINT``: Metadata API endpoint URL.
+        [defaults]
+        token = ABC-123456789123456789123456789
 
-    ``DWAVE_API_TOKEN``: API authorization token.
+        [default-solver]
+        solver = {"qpu": true, "num_qubits__gt": 5000}
 
-    ``DWAVE_API_SOLVER``: Default solver.
+        [hybrid]
+        client = hybrid
```

Comment:
Probably more useful config would be:
```suggestion
        [bqm]
        client = hybrid
        solver = {"supported_problem_types__contains": "bqm"}

        [cqm]
        client = hybrid
        solver = {"supported_problem_types__contains": "cqm"}
```
(`client` is optional above, but can leave it for explicitness)

Then:
```
>>> with Client.from_config(profile="cqm") as client:
...     cqm_solver = client.get_solver()
```